pymbar/mbar.py

```diff
@@ -301,7 +302,33 @@ def __init__(self, u_kn, N_k, maximum_iterations=10000, relative_tolerance=1.0e-
                 # which might involve passing in different combinations of options, and passing out other strings.
                 solver['options']['verbose'] = self.verbose
 
-        self.f_k = mbar_solvers.solve_mbar_for_all_states(self.u_kn, self.N_k, self.f_k, solver_protocol)
+        np.random.seed(rseed)
+        self.rstate = np.random.get_state()
+
+        f_k_init = self.f_k.copy()  # we need to pass a copy so we don't overwrite the original 
+        self.f_k = mbar_solvers.solve_mbar_for_all_states(self.u_kn, self.N_k, f_k_init, solver_protocol)
+
+        self.nbootstraps = None
+        if nbootstraps != None:
+            self.nbootstraps = nbootstraps
+            #save the original data:
+            self.f_k_boots = np.zeros([self.nbootstraps,self.K])
+
+            allN = int(np.sum(self.N_k))
+            rinit = np.zeros(allN, int)
```

Comment:
@mrshirts : This line should probably be moved to be inside `for b in range(self.nbootstraps):` but outside ` for k in range(K):` right? It's also probably fine where it is.

Reply:
It should be fine where it is, because all of the states have some value of K, so rinit gets completely refilled each iteration